src/XrdOuc/XrdOucUtils.cc

```diff
@@ -1409,5 +1411,77 @@ void XrdOucUtils::trim(std::string &str) {
     while( str.size() && !isgraph(str[str.size()-1]) )
         str.resize (str.size () - 1);
 }
+
+std::string XrdOucUtils::obfuscate(const std::string &input, const std::string &key, const char delimiter) {
+  size_t i = 0;
+  std::string ret;
+  while(i < input.length() && i < key.length()) {
+    if(std::tolower(input[i]) == key[i]) {
+      ret += input[i];
+    } else {
+      // at least one character is different, no key match so no obfuscation needed
+      return input;
+    }
+    i++;
+  }
+  if(i == key.length()) {
+    while(i < input.length() && input[i] != delimiter) {
+      if(std::isspace(input[i])) {
+        ret += input[i];
+      } else {
+        return input;
+      }
+      i++;
+    }
+    //Spaces between the key and the delimiter got trimmed
+    if(i < input.length() && input[i] == delimiter) {
+      ret += input[i];
+      ret += " ***********";
```

Comment:
I think we should follow Brian's suggestion of "****REDACTED**** as the substitiion string.